src/main/resources/config/application-dev.properties

```diff
@@ -0,0 +1,42 @@
+# Logging level for the application loggers (all built-in routes and classes)
```

Comment:
I'm moving [this - application.properties](https://github.com/ozone-his/ozone-distro/blob/main/distro/configs/eip-odoo-openmrs/properties/application.properties) to be part of this project, by doing this, the properties aren't friendly for test environment because the env vars placeholders(for production environment). To solve this, we use Spring Profiles(Introducing profile-specific properties).

If you're concerned about overriding the `application.properties` file, you can achieve the same outcome by overriding the `application-prod.properties` file instead.

Note: The overriding of property values happens through environment variables, Docker overrides the `application.properties`.

Why do you want to maintain one `application.properties` file ??

Reply:
> Why do you want to maintain one application.properties file ??

Maintaining the fewest number of files is not only precisely cleaner but also reduces the overhead of having to update multiple files thus reducing duplicates.

> If you're concerned about overriding the application.properties file, you can achieve the same outcome by overriding the application-prod.properties file instead.

This would mean maintain one of the properties file instead of both?

Reply:
> This would mean maintain one of the properties file instead of both?

You can't avoid having multiple files when using Spring Profiles.
 How would you handle the properties for different environments? For this case `test` and `prod`

Reply:
`spring.profiles.active=test,prod,dev` inside the maintained properties file would suffice right?

Reply:
Yes if we had Profile-specific beans/configuration which we don't, and to do that we have to modify `openmrs-eip` project. An alternative is using Profile-specific Properties Files which is what I'm doing.

`spring.profiles.active=test,prod,dev` activates the 3 profiles but first, you will need to create the three files; `application-test.properties`, `application-prod.properties`, `application-dev.properties` with profile-specific properties. In the end, you have 4 files.

Reply:
I do understand the motivation here. I'm not opposed to the idea especially if this will not cause confusion/ambiguity. I noticed a few varying property values between the different profiles. IMO the profile based approach is somewhat overkill and adds some unnecessary complexity and/or redundancy. In the end we'd be basically shifting some docker compose functionality to be handled at Spring level. 

On the other hand, If it were possible too, I'd have one property file to act as the single source of truth in terms of defined properties to configure, and have that re-used elsewhere. I find it a little not straight forward seeing a number of files with similar properties but different values for different profiles.

Reply:
The whole point of profiles, though, is to allow us, for example to say "on CI the value for `spring.mngt-datasource.driverClassName` is `org.h2.Driver`". If the issue here is about the number of properties, we can probably look into simplifying that, e.g., I'm not sure why we need `emr.weight.concept=` in the dev profile here since the default value `${EMR_WEIGHT_CONCEPT}` seems like to resolve to an empty string anyway.

Reply:
> we can probably look into simplifying that

Yes indeed. This @corneliouzbett has tried to simplify already by reducing the number of files involved.